### PR TITLE
Enable/disable location share button when setting is changed

### PR DESCRIPTION
**Pull request by andybalaam**
immediate-loc-share

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://61e185e16a0e260b6b33d67a--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/rooms/MessageComposer.tsx [hunk 1]
```diff
@@ -344,6 +347,15 @@ export default class MessageComposer extends React.Component<IProps, IState> {
                         }
                         break;
                     }
+
+                    case "feature_location_share": {
+                        const showLocationButton = SettingsStore.getValue(
+                            "feature_location_share");
```
Comment (t3chguy) [inline src/components/views/rooms/MessageComposer.tsx:353]: this looks like it would fit on one line

**Review by t3chguy — APPROVED** [review]
LGTM otherwise

File: src/components/views/rooms/MessageComposer.tsx (hunk 1 above)
Comment (andybalaam) [reply src/components/views/rooms/MessageComposer.tsx:353]: Based on "try to keep JavaScript code around the 90 column mark" from the [style doc](https://github.com/matrix-org/matrix-react-sdk/blob/master/code_style.md#general-style) (this is 100 columns), I'd vote for wrapping it.

Comment (andybalaam) [reply src/components/views/rooms/MessageComposer.tsx:353]: Of course, if I had my way we'd have a hard limit at 80 columns except imports and JSX...

**Pull request merged**